openwisp_users/base/models.py

```diff
@@ -129,6 +129,17 @@ def clean(self):
         if self.phone_number == '':
             self.phone_number = None
 
+    @cached_property
+    def get_permissions(self):
```

Comment:
if it's a property, we don't need to name it `get_permissions`, we can name it just `permissions`.

The other problem is that `cached_property` does not cache using the cache storage, it caches only in memory, so the query will be executed again at subsequent requests.

We have to do something like `organizations_dict`: cache it in the cache storage (one key for each user) and invalidate the cache whenever the permissions or the groups of a user are changed.

Please study this PR and try to replicate what has been done there: https://github.com/openwisp/openwisp-users/pull/134/files